ulid2.py

```diff
@@ -189,7 +192,13 @@ def generate_binary_ulid(timestamp=None):
     ts_bytes = _to_binary(
         (ts >> shift) & 0xFF for shift in (40, 32, 24, 16, 8, 0)
     )
-    return ts_bytes + os.urandom(10)
+    entropy = os.urandom(10)
+    if _last_timestamp == ts and _last_entropy is not None:
```

Comment:
I'm a little concerned about thread/re-entrancy safety here, what with global state and all.

I see the two options being

* adding a note that multiple threads generating ulids may not exhibit the same monotonicity behavior
* adding a lock around all of this

and I'd rather not add a lock, since many (most?) Python apps are single-threaded anyway.

Reply:
I added a note to the README in my PR. Personally no issues with the lock besides the ~irrelevant performance hit and the added complexity, let me know if you'd rather have it locked!